nova/utils.go

```diff
@@ -0,0 +1,105 @@
+package nova
+
+import (
+	"encoding/json"
+	"fmt"
+	"gopkg.in/yaml.v2"
+	"io"
+	"io/ioutil"
+	"net"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/rackspace/gophercloud"
+	"github.com/rackspace/gophercloud/openstack"
+	"github.com/stackanetes/kubernetes-entrypoint/logger"
+)
+
+func loadConfigs(confPath string) (config map[string]string, err error) {
+	absFP, err := filepath.Abs(confPath)
+	if err != nil {
+		logger.Error.Println()
+		return config, fmt.Errorf("Cannot retrieve absolute path for config file.")
+	}
+
+	yamlFile, err := ioutil.ReadFile(absFP)
+	if err != nil {
+		return config, fmt.Errorf("Cannot load config file.")
+	}
+	if err = yaml.Unmarshal(yamlFile, &config); err != nil {
+		return config, fmt.Errorf("Invalid format of config file.")
+	}
+	logger.Info.Printf("Configuration loaded from %s.\n", absFP)
+
+	return
+}
+
+func getJson(body io.ReadCloser, target interface{}) error {
+	defer body.Close()
+
+	return json.NewDecoder(body).Decode(target)
+}
+
+func createOpenstackClient(confPath string) (client *gophercloud.ServiceClient, err error) {
+	config, err := loadConfigs(confPath)
+	if err != nil {
+		return nil, fmt.Errorf("Cannot load variables required to create openstack client.")
+	}
+
+	ao := gophercloud.AuthOptions{
+		IdentityEndpoint: config["IdentityEndpoint"],
+		Username:         config["Username"],
+		Password:         config["Password"],
+		TenantName:       config["TenantName"],
+		DomainID:         config["DomainID"],
+	}
+
+	provider, err := openstack.AuthenticatedClient(ao)
+	if err != nil {
+		return client, fmt.Errorf("Cannot create openstack provider: %v", err)
+	}
+	for a := 1; a < 4; a++ {
```

Comment:
4 should be somehow configurable, we should have a global api retry number variable (obtained from argv).